src/activities/morse_code/Morse_code.qml

```diff
@@ -0,0 +1,438 @@
+                width: 1
+                height: 5 * ApplicationInfo.ratio
+            }
+            GCText {
+                id: feedback
+                anchors.horizontalCenter: parent.horizontalCenter
+                text: items.toAlpha ?
+                          qsTr("Morse value: %1").arg(value) :
+                          qsTr('Alphabet/Numeric value: %1').arg(value)
+                onTextChanged: timer.start()
+                color: 'white'
+
+                Rectangle {
+                    z: -1
+                    opacity: 0.8
+                    gradient: Gradient {
+                        GradientStop { position: 0.0; color: "#000" }
+                        GradientStop { position: 0.9; color: "#666" }
+                        GradientStop { position: 1.0; color: "#AAA" }
+                    }
+                    radius: 10
+                    border.color: 'black'
+                    border.width: 1
+                    anchors.centerIn: parent
+                    width: parent.width * 1.1
+                    height: parent.height
+                }
+                property string value: items.toAlpha ?
+                                           morseConverter.morse.trim() :
+                                           morseConverter.alpha ? morseConverter.alpha : ''
+            }
+
+            Timer {
+                id: timer
+                interval: 500
+                onTriggered: {
+                    if( feedback.value == items.questionValue )
+                        bonus.good('tux')
+                }
+            }
+            Item {
+                id:margin
+                width: 1
+                height: 20 * ApplicationInfo.ratio
+            }
+
+            GCText {
+                id: instruction
+                wrapMode: TextEdit.WordWrap
+                fontSize: mediumSize
+                horizontalAlignment: Text.AlignHCenter
+                anchors.horizontalCenter: parent.horizontalCenter
+                width: parent.width * 0.5
+                height: parent.height * 0.9
+                text: items.instruction
+
+                color: 'white'
+                Rectangle {
+                    z: -1
+                    opacity: 0.8
+                    gradient: Gradient {
+                        GradientStop { position: 0.0; color: "#000" }
+                        GradientStop { position: 0.9; color: "#666" }
+                        GradientStop { position: 1.0; color: "#AAA" }
+                    }
+                    radius: 10
+                    border.color: 'black'
+                    border.width: 1
+                    width: parent.width
+                    height: parent.height * 0.5
+                }
+            }
+
+
+        }
+
+        Score {
+            id: score
+            anchors.bottom: keyboard.top
+            currentSubLevel: 0
+            numberOfSubLevels: 1
+            visible: !(first_screen)
```

Comment:
!first_screen.visible instead? If the number of sublevels is not fixed, there is no need to put a default value here